sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFramesSuite.scala

```diff
@@ -548,4 +548,19 @@ class DataFrameWindowFramesSuite extends QueryTest with SharedSparkSession {
       df,
       Row(1) :: Row(1) :: Nil)
   }
+
+  test("SPARK-45352: Remove window partition if partition expression are foldable") {
```

Comment:
for a new optimizer rule, we should add unit tests, like `LimitPushdownSuite`

Reply:
Done